### PR TITLE
Leaflet: restructured into UMD with satellite modules as augmentations

### PR DESCRIPTION
**Pull request by aluanhaddad**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

I'm not 100% sure if this is the correct way to model the dependencies between Leaflet and the many surrounding packages that enrich it. 
There are a number of issues with how Leaflet is currently represented.
It is an ambient external module.
It depends on a global but that is not necessary
In the process of trying to change these, it soon became clear that adjusting ~11 other packages would be necessary.

I am curious to see the feedback on this PR, especially as there are many longstanding libraries, like jQuery, that still have not migrated to the UMD declaration format, it will be interesting to see if this is deemed practical.

I am also keen to learn if there is a better approach to take here.

Thanks!

### CODE REVIEW

File: types/leaflet.gridlayer.googlemutant/index.d.ts [hunk 1]
```diff
@@ -1,77 +1,79 @@
-// Type definitions for leaflet.gridlayer.googlemutant 0.4
+// Type definitions for leaflet.gridlayer.googlemutant 0.5
 // Project: https://gitlab.com/IvanSanchez/Leaflet.GridLayer.GoogleMutant#README
 // Definitions by: Ernest Rhinozeros <https://github.com/ernest-rhinozeros>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-/// <reference types="leaflet" />
+import L = require('leaflet');
```
Comment (DenisCarriere) [inline types/leaflet.gridlayer.googlemutant/index.d.ts]: I don't think this should be using `require`.

**Should be**

```javascript
import * as L from 'leaflet'
```

File: types/leaflet.locatecontrol/index.d.ts [hunk 2]
```diff
@@ -3,9 +3,9 @@
 // Definitions by: Denis Carriere <https://github.com/DenisCarriere>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-/// <reference types="leaflet" />
+import L = require('leaflet');
 
-declare namespace L {
```
Comment (DenisCarriere) [inline types/leaflet.locatecontrol/index.d.ts]: This breaks the tests, the Type definition should be defined in the `namespace L`.

image

**Review by DenisCarriere — CHANGES_REQUESTED** [review]

**Comment by DenisCarriere** [comment]
Big no 👎, this PR won't pass any tests soon.

**Comment by paulvanbrenk** [comment]
I would break this into smaller PRs or at least commits and start with changes to the main leaflet package to get that updated.

That way it's easier to review, and help you get to a good end result. As is this PR is very intimidating.

**Comment by paulvanbrenk** [comment]
Going to close this PR @DenisCarriere

**Pull request closed without merging**